server/src/main/java/org/elasticsearch/action/admin/cluster/remote/RemoteClusterNodesAction.java

```diff
@@ -43,13 +44,25 @@ public RemoteClusterNodesAction() {
     }
 
     public static class Request extends ActionRequest {
+        private final boolean remoteClusterServer;
 
-        public static final Request INSTANCE = new Request();
+        public Request() {
+            this.remoteClusterServer = false;
+        }
 
-        public Request() {}
+        public Request(boolean remoteClusterServer) {
+            this.remoteClusterServer = remoteClusterServer;
+        }
 
         public Request(StreamInput in) throws IOException {
             super(in);
+            this.remoteClusterServer = in.readBoolean();
+        }
+
+        @Override
+        public void writeTo(StreamOutput out) throws IOException {
+            super.writeTo(out);
+            out.writeBoolean(remoteClusterServer);
```

Comment:
Strictly speaking, we should add a TransportVersion for this change. But it seems rather wasteful since the feature is not released.